Makefile.core.mk

```diff
@@ -202,17 +202,22 @@ endif
 # List of all binaries to build
 # We split the binaries into "agent" binaries and standard ones. This corresponds to build "agent".
 # This allows conditional compilation to avoid pulling in costly dependencies to the agent, such as XDS and k8s.
-AGENT_BINARIES:=./pilot/cmd/pilot-agent ./cni/cmd/istio-cni
```

Comment:
I think moving it from AGENT_BINARIES will mean the binary size gets bloated a lot. Its basically "trim dependencies" mode

Reply:
@howardjohn Sorry, can you explain that a bit more? I don't see where AGENT_BINARIES are handled differently than anything else, but I might be missing it.

Reply:
AGENT_BINARIES build with `agent` tag which trims dependencies

Reply:
@howardjohn saw that and assumed it wasn't relevant because none of the CNI stuff has that tag, but of course it has deps that do.

It's a little inelegant but should be fixed - I would rather not get into OS detection via makefile if I can avoid it.